src/mpl2/src/hier_rtlmp.cpp

```diff
@@ -1733,24 +1822,30 @@ void HierRTLMP::breakLargeFlatCluster(Cluster* parent)
   // vertices
   // other clusters behaves like fixed vertices
   // We do not consider vertices only between fixed vertices
-  int num_vertices = cluster_map_.size() - 1;
+  int num_vertices = cluster_map_.size();
   num_vertices += parent->getLeafMacros().size();
   num_vertices += std_cells.size();
   int vertex_id = 0;
+  std::vector<float> vertex_weight;
   for (auto& [cluster_id, cluster] : cluster_map_) {
     cluster_vertex_id_map[cluster_id] = vertex_id++;
+    vertex_weight.push_back(0.0f);
   }
   for (auto& macro : parent->getLeafMacros()) {
     inst_vertex_id_map[macro] = vertex_id++;
+    const sta::LibertyCell* liberty_cell = network_->libertyCell(macro);
+    vertex_weight.push_back(liberty_cell->area());
   }
   int num_fixed_vertices = vertex_id;  // we set these fixed vertices to part0
   for (auto& std_cell : std_cells) {
     inst_vertex_id_map[std_cell] = vertex_id++;
+    const sta::LibertyCell* liberty_cell = network_->libertyCell(std_cell);
+    vertex_weight.push_back(liberty_cell->area());
   }
-  std::vector<int> part(num_vertices, -1);
-  for (int i = 0; i < num_fixed_vertices; i++) {
-    part[i] = 0;
-  }
+  // std::vector<int> part(num_vertices, -1);
+  // for (int i = 0; i < num_fixed_vertices; i++) {
+  //  part[i] = 0;
+  //}
```

Comment:
Remove - I'll stop mentioning it.  Please do it throughout (and on all future PRs as I'm always giving this same feedback).